ryhope/src/error.rs

```diff
@@ -0,0 +1,66 @@
+use thiserror::Error;
+use tokio_postgres::error::Error as PgError;
+
+#[derive(Error, Debug)]
+pub enum RyhopeError {
+    /// An error that occured while interacting with the DB.
+    #[error("DB error while {msg}: {err}")]
+    DbError { msg: String, err: PgError },
+
+    /// An error that occured while interacting with the DB.
+    #[error("DB error while {msg}: {err}")]
+    DbPoolError {
+        msg: String,
+        err: bb8::RunError<PgError>,
+    },
```

Comment:
what's the difference between the two ? i.e. not sure we can act differently against one variant vs another.

Reply:
One is an issue stemming from tokio_postgres, the other one from bb8, _e.g._ when failing to acquire a connection from the pool.

> i.e. not sure we can act differently against one variant vs another.

Neither am I to be frank, I think they stand at the same level of seriousness.

Reply:
In that case i would put them under the same DB error tbh, and put a big comment that it's unrecoverable and something's wrong with the DB or something.

Reply:
I can't, because they don't wrap the same error type. Or you would like it to wrap an enum that could be either error?